### ISSUE TITLE
Reconnecting after connection loss

### ISSUE THREAD

**Issue by jp1337** [Dichloromethane/skat#4]
You can successfully connect to the server once.
When you kill the client application or when the network is disconnected you can not reconnect to the server with the same player name.

Server log:
```
Nov 22 12:34:00 root2 skat[5585]: DEBUG (server_listener:310)
Nov 22 12:34:00 root2 skat[5585]:      Received connection 4
Nov 22 12:34:00 root2 skat[5585]: DEBUG (server_handler:264)
Nov 22 12:34:00 root2 skat[5585]:      Handler started, establishing connection with 4
Nov 22 12:34:00 root2 skat[5585]: DEBUG (establish_connection_server:151)
Nov 22 12:34:00 root2 skat[5585]:      New player join with name 'JP'
Nov 22 12:34:00 root2 skat[5585]: DEBUG (establish_connection_server:157)
Nov 22 12:34:00 root2 skat[5585]:      Existing player 0: 'JP' (inactive)
Nov 22 12:34:00 root2 skat[5585]: DEBUG (establish_connection_server:159)
Nov 22 12:34:00 root2 skat[5585]:      Empty player slot 1
Nov 22 12:34:00 root2 skat[5585]: DEBUG (establish_connection_server:159)
Nov 22 12:34:00 root2 skat[5585]:      Empty player slot 2
Nov 22 12:34:00 root2 skat[5585]: DEBUG (establish_connection_server:159)
Nov 22 12:34:00 root2 skat[5585]:      Empty player slot 3
Nov 22 12:34:00 root2 skat[5585]: #033[31mERROR#033[0m (conn_error:90)
Nov 22 12:34:00 root2 skat[5585]:      Connection error: CONN_ERROR_PLAYER_NAME_IN_USE
Nov 22 12:34:00 root2 skat[5585]: DEBUG (send_package:38)
Nov 22 12:34:00 root2 skat[5585]:      Sending package of type PACKAGE_ERROR with payload size 4
Nov 22 12:34:00 root2 skat[5585]: DEBUG (server_handler:267)
Nov 22 12:34:00 root2 skat[5585]:      Establishing connection with 4 failed
```

Client log:
```
sensitive_data_here$ ./skat_client -h root2.webdesign-kronberg.de JP
Options: host=root2.webdesign-kronberg.de; port=55555; name=JP; resume=0; graphical=0
DEBUG (client_init:499)
     Initializing client 'JP' for server 'root2.webdesign-kronberg.de:55555'
DEBUG (ctimer_create:19)
     Creating timer
DEBUG (client_signal_handler:448)
     Starting interrupt signal handler thread
DEBUG (client_run:523)
     Running client with with connection mode 'new'
DEBUG (start_client_conn:152)
     Established connection on socket 4
DEBUG (establish_connection_client:323)
     Establishing new connection to server
DEBUG (ctimer_run:49)
     Setting timer to 500000000
DEBUG (handle_console_input:646)
     Started console input thread
> DEBUG (send_package:37)
     Sending package of type PACKAGE_JOIN with payload size 19
DEBUG (ctimer_run:58)
     Started Timer
ERROR (establish_connection_client:366)
     Encountered error CONN_ERROR_PLAYER_NAME_IN_USE while connecting to server
ERROR (client_conn_thread:86)
     Could not establish connection to server, exiting
```

**Comment by iTitus** [comment]
When using the `-r` flag the client will try to reconnect. Then it should work.
But I would advise against trying to disconnect and then reconnect during a game, that is not handled gracefully yet.

**Comment by iTitus** [comment]
Reconnencting does not work

**Comment by jp1337** [comment]
When using `-r` the error is a different one: `CONN_ERROR_NO_SUCH_PLAYER_NAME`

```
sensitive_data_here$ ./skat_client -r -h root2.webdesign-kronberg.de JP
Options: host=root2.webdesign-kronberg.de; port=55555; name=JP; resume=1; graphical=0
DEBUG (client_init:499)
     Initializing client 'JP' for server 'root2.webdesign-kronberg.de:55555'
DEBUG (ctimer_create:19)
     Creating timer
DEBUG (client_signal_handler:448)
     Starting interrupt signal handler thread
DEBUG (client_run:523)
     Running client with with connection mode 'resume'
DEBUG (start_client_conn:152)
     Established connection on socket 4
DEBUG (establish_connection_client:323)
     Resuming connection to server
DEBUG (ctimer_run:49)
     Setting timer to 500000000
DEBUG (handle_console_input:646)
     Started console input thread
> DEBUG (send_package:37)
     Sending package of type PACKAGE_CONN_RESUME with payload size 19
DEBUG (ctimer_run:58)
     Started Timer
ERROR (establish_connection_client:366)
     Encountered error CONN_ERROR_NO_SUCH_PLAYER_NAME while connecting to server
ERROR (client_conn_thread:86)
     Could not establish connection to server, exiting
```

**Comment by iTitus** [comment]
Fixed in 320336b1f490bbf89ea8bf5dcac4b909d0906047
